internal/observability-operator/helm_test.go

```diff
@@ -0,0 +1,131 @@
+package observabilityoperator
+
+import (
+	"context"
+	"testing"
+
+	operatorsv1 "github.com/operator-framework/api/pkg/operators/v1"
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/rhobs/multicluster-observability-addon/internal/addon"
+	"github.com/rhobs/multicluster-observability-addon/internal/observability-operator/handlers"
+	"github.com/rhobs/multicluster-observability-addon/internal/observability-operator/manifests"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"open-cluster-management.io/addon-framework/pkg/addonfactory"
+	"open-cluster-management.io/addon-framework/pkg/addonmanager/addontesting"
+	"open-cluster-management.io/addon-framework/pkg/agent"
+	addonapiv1alpha1 "open-cluster-management.io/api/addon/v1alpha1"
+	fakeaddon "open-cluster-management.io/api/client/addon/clientset/versioned/fake"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+var (
+	_ = operatorsv1.AddToScheme(scheme.Scheme)
+	_ = operatorsv1alpha1.AddToScheme(scheme.Scheme)
+)
+
+func fakeGetValues(ctx context.Context, k8s client.Client) addonfactory.GetValuesFunc {
+	return func(
+		_ *clusterv1.ManagedCluster,
+		mcAddon *addonapiv1alpha1.ManagedClusterAddOn,
+	) (addonfactory.Values, error) {
+		opts := handlers.BuildOptions(ctx, k8s, mcAddon, addon.ObservabilityOperatorOptions{Enabled: true})
```

Comment:
Here we have to get the AddOnDeploymentConfig otherwise we are not actually testing the signal/feature end to end  
```suggestion
		aodc := &addonapiv1alpha1.AddOnDeploymentConfig{}
		keys := addon.GetObjectKeys(mcAddon.Status.ConfigReferences, addonutils.AddOnDeploymentConfigGVR.Group, addon.AddonDeploymentConfigResource)
		if err := k8s.Get(context.TODO(), keys[0], aodc, &client.GetOptions{}); err != nil {
			return nil, err
		}
		addonOpts, err := addon.BuildOptions(aodc)
		if err != nil {
			return nil, err
		}

		opts := handlers.BuildOptions(ctx, k8s, mcAddon, addonOpts.Platform.ObservabilityOperator)
```